ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/panelcontainer/v1/panelcontainer/clientlibs/editor/js/saveasfragment.js

```diff
@@ -0,0 +1,208 @@
+        const fragments = [];
+        components.forEach(component => {
+            const componentPath = component.getPath();
+            const compTemplatePath = component.getTemplatePath();
+            if (compTemplatePath && !fragmentCmpCache.hasOwnProperty(componentPath)) {
+                const templateJson = getComponentJson(compTemplatePath, ".json");
+                fragmentCmpCache[componentPath] = templateJson?.fieldType === "panel" && templateJson.hasOwnProperty("fragmentPath");
+            }
+            if(fragmentCmpCache[componentPath]) {
+                fragments.push(component)
+            }
+        });
+        return fragments;
+    }
+
+    function getComponentJson(componentPath, jsonPath) {
+        const result = $.ajax({
+            type: 'GET',
+            async: false,
+            url: Granite.HTTP.externalize(componentPath + jsonPath),
+            cache: false
+        });
+        return result.responseJSON;
+    }
+
+    const initializeFragmentSelector = function (dialog, fragments) {
+        const fragmentSelect = dialog.querySelector(".cmp-adaptiveform-saveasfragment__fragmentselector-container");
+
+        var label = '<label class="coral-Form-fieldlabel">' + Granite.I18n.get("Select Fragment Component*") + '</label>';
+        var autocomplete = new Coral.Autocomplete().set({
+            placeholder: Granite.I18n.getMessage("Select Fragment"),
+        });
+        autocomplete.name = "fragmentComponent"
+        autocomplete.required = true;
+        fragments.forEach((fragment, idx) => {
+            autocomplete.items.add({
+                value: fragment.getResourceType(),
+                content: {
+                    textContent: fragment.getTitle(),
+                    cmpIndex: idx
+                },
+                selected: idx === 0
+            })
+        })
+        autocomplete.classList.add("coral-Form-field");
+        fragmentSelect.innerHTML = label;
+        fragmentSelect.appendChild(autocomplete);
+    }
+
+    const initializeDataModel = function (dialog, editable) {
+        const schemaType = containerProperties.schemaType;
+        const schemaRef = containerProperties.schemaRef;
+        const schemaTypeElement = dialog.querySelector(FORM_MODEL_SELECTOR);
+        if(schemaType) {
+            schemaTypeElement && (schemaTypeElement.value = schemaType);
+        }
+        if(schemaRef) {
+            const schemaRefElement = dialog.querySelector(SCHEMA_REF);
+            schemaRefElement && (schemaRefElement.value = schemaRef);
+        }
+        schemaTypeElement.disabled = true;
+        window.CQ.FormsCoreComponents.Utils.DataModel.initialiseDataModel($(dialog));
+    }
+
+    const initializeTargetPath = function(dialog, editable) {
+        const path = editable.path;
+        const targetPathElement = dialog.querySelector("[name='targetPath']");
+        if(path.startsWith("/content/forms/af/") && path.indexOf("/jcr:content/guideContainer/") > -1) {
```

Comment:
Will this work is panel is inside SITES, please cross check ? let's create constants for this

Reply:
Yes this will work for Sites also, will create constant !!